Source/MQTTnet/MqttTopicFilterComparer.cs

```diff
@@ -174,5 +174,19 @@ public static unsafe MqttTopicFilterCompareResult Compare(string topic, string f
 
             return MqttTopicFilterCompareResult.NoMatch;
         }
+
+        public static bool ContainsWildcards(string topicFilter)
+        {
+            for (var i = 0; i < topicFilter.Length; i++)
+            {
+                var c = topicFilter[i];
+                if (c == MultiLevelWildcard || c == SingleLevelWildcard)
```

Comment:
The MultiLevelWildcard (#) can only be at the end of the topic. So, we can check this once if the topic is at least 3 chars long (the char before the # must be a / in all cases).